components/scream/src/physics/p3/p3_upwind_impl.hpp

```diff
@@ -106,7 +106,7 @@ ::generalized_sedimentation (
 {
   // compute dt_sub
   const Int tmpint1 = static_cast<int>(Co_max + 1);
-  const Scalar dt_sub = ekat::impl::min(dt_left, dt_left/tmpint1);
+  const Scalar dt_sub = tmpint1 == 0 ? dt_left : ekat::impl::min(dt_left, dt_left/tmpint1);
```

Comment:
P3 upwind change is here. Would like to get some eyes on this for sure.

Reply:
Courant number. Co_max should be >= 0 on input. I believe all the callers to generalized_sedimentation have code that assures Co_max >= 0. Is it just a unit test that causes tempint1 0?

Separately, what Luca writes is correct: that min doesn't make sense. Seems to have been in the F90 from the start, so it was just copied over.

Taken together, here you could write simply
```C++
assert(Co_max >= 0);
const Scalar dt_sub = dt_left/tmpint1;
```

Reply:
@ambrad , yes. I think this problem came up in one of the random-input bfb tests.

Reply:
Then maybe there is some random input that needs better constraints for the pdf?